### DISCUSSION TITLE
Login Use Case: How to login via google to my website or perform user authentication for my website?

### DISCUSSION THREAD

**Discussion by Himanshu21git** [webdriverio/webdriverio#11256] [Q&A]
Hi,

I am currently trying to write some automated tests for my website. But most of the functionality that I want to test, is available only after the user logs in to my website.

How to perform this login operation using Webdriverio and make the user get logged in to my website?

Need your help on how to resolve this issue.
I want the user to login to my website during or before the test run, and perform the desired tests on the pages that require the user to be authenticated.

**Comment by christian-bromann** [comment]
> How to perform this login operation using Webdriverio and make the user get logged in to my website?

You can always set a cookie to have the user be authenticated. How this works in detail is highly dependent on what type of authentication you have.

**Reply by Himanshu21git** [reply]
Thanks for the reply. Will explore this and revert back.
